weekly_report/generate_report.py

```diff
@@ -229,6 +244,37 @@ def get_website_names(self) -> list[str]:
                 website_list.append(websites[x]['website_name'])
             return website_list
 
+    def get_word_cloud(self):
```

Comment:
Very chunky function at first glance; would it possible to delegate functionality across multiple functions?

Reply:
Tried to address